tests/components/hassio/test_http.py

```diff
@@ -147,3 +149,12 @@ async def test_snapshot_upload_headers(hassio_client, aioclient_mock):
 
     req_headers = aioclient_mock.mock_calls[0][-1]
     req_headers["Content-Type"] == content_type
+
+
+def test_need_auth(hass):
+    """Test if the requested path needs authentication."""
+    assert not _need_auth(hass, "addons/test/logo")
+    assert _need_auth(hass, "snapshots/new/upload")
+
+    hass.data["onboarding"] = False
+    assert not _need_auth(hass, "snapshots/new/upload")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)